src/main/webapp/js/datepicker.js

```diff
@@ -70,8 +70,12 @@ function triggerDatepickerOnClick(datepickerDivs) {
 }
 
 /**
+ * Compares startDate and publishDate and returns the earlier date as minDate,
+ * if publishDate is undefined or null, returns startDate as minDate.
  * @assumption: startDate has a valid value
- * @returns {Date} publishDate if it is valid and smaller than startDate, else startDate
+ * @param {Date} startDate
+ * @param {Date} publishDate
+ * @returns {Date}
  */
 function getMaxDateForVisibleDate(startDate, publishDate) {
```

Comment:
@Neurrone i am also getting a bit confused,
`$('#publishdate').datepicker({
        dateFormat: 'dd/mm/yy',
        showOtherMonths: true,
        gotoCurrent: true,
        defaultDate: tomorrow,
        onSelect: function() {
            var newVisibleDate = getMaxDateForVisibleDate($('#startdate').datepicker('getDate'),
                    $('#publishdate').datepicker('getDate'));
            $('#visibledate').datepicker('option', 'maxDate', newVisibleDate);
        }
    });`

Here,I think using the function getMaxDateForVisibleDate we are finding the minimum time from which the session can be visible 
Also i don't think there would be any case where startDate > publishDate ,is it?